src/python/turicreate/toolkits/_model.py

```diff
@@ -636,7 +636,7 @@ def _load_version(cls, state, version):
     def __init__(self):
         pass
 
-    def name(self):
+    def _name(self):
         """
```

Comment:
Doesn't this effect public API. I think the right way to fix this would be to deprecate this for 1 release and then remove it for the next release. https://docs.python.org/3/library/exceptions.html#DeprecationWarning.

@TobyRoseman Thoughts?

Reply:
That sounds good to me. Also please include a warning in the docstring.

Taking a look at #2321 will probably be helpful.

Reply:
Very good!

Reply:
Oh but somehow the models which rely on C++ model (AC,OD,SC...) the `name()` function are already gone in `6.0`.
@TobyRoseman @srikris

Reply:
That's okay, But this is public API for all other models so let's follow the deprecation procedure and then file a bug to fix it in 6.2

Reply:
Good!

Reply:
Good!